### ISSUE TITLE
Ability to pass previous source map content

### ISSUE THREAD

**Issue by yuri-karadzhov** [sass/node-sass#2860]
At the moment there is no possibility to pass previous source map content it would be nice if `sourceMap` option will be able to receive previous source map object and use it to generate new source map.
```js
sass.render({
      data,
      sourceMap: prevSourceMapObject
}, handleResult);
```

**Comment by saper** [comment]
How should it work?

**Comment by yuri-karadzhov** [comment]
Same way as with other compilers e.g. postcss I guess. If your `scss` is result of precompilation and has source map content already it is pity to lose it after compilation with node-sass. Actually it works with postcss-node-sass, but it is hacky and postcss people recommend to use node-sass directly.

**Comment by nschonni** [comment]
I don't think this is something for node-sass/libsass. PostCSS is a good idea, but node-sass is the start of the compilation chain, there is nothing that Sass in general would do with existing source maps coming from things like plain CSS files. Pass the compiled node-sass and any other CSS you're trying to concatenate to PostCSS that already knows how to combine source maps

**Comment by hqjs** [comment]
Well, in my case https://hqjs.org it is not, I have additional step before node-sass to make it work with paths. So far it does not change files significantly so should be Ok without source map, but I can assume there might be some useful plugins that transform scss into scss so it might become a thing at some point.

**Comment by saper** [comment]
Would that be possible to create a simple demo repository to demonstrate the case - and most importantly to show what the expected output would be?

**Comment by mgreter** [comment]
IMO this is a feature libsass already has on the [wish-list][1], but it is prioritized very very low.
The related ticket would be https://github.com/sass/libsass/issues/668
And https://github.com/sass/node-sass/issues/1584 seems to be a duplicate of this.

[1]: https://github.com/sass/libsass/issues/1892#issue-128067430
